modules/build/src/test/scala/scala/build/tests/ScalaPreprocessorTests.scala

```diff
@@ -0,0 +1,83 @@
+package scala.build
+package tests
+
+import scala.build.preprocessing.ScalaPreprocessor
+import scala.build.Logger
+import com.eed3si9n.expecty.Expecty.{assert => expect}
+import _root_.scala.build.errors.Diagnostic
+
+class ScalaPreprocessorTests extends munit.FunSuite {
+
+  val path = os.pwd
+
+  trait Check {
+    def test(diags: Seq[Diagnostic]): Boolean
+  }
+  case class Warn(messagePattern: String, line: Int, index: Int) extends Check {
+    def test(diags: Seq[Diagnostic]): Boolean = {
+      val matching = diags.filter(_.positions.collect {
+        case Position.File(_, (l, c), _) if l == line && index == c => l
```

Comment:
NIT
```suggestion
        case Position.File(_, (`line`, `index`), _) => line
```

(That said, I often forget about matching with backticks myself, so there might be other cases in the code base where we can do the same…)